ee/server/service/teams.go

```diff
@@ -391,6 +391,24 @@ func (svc *Service) DeleteTeam(ctx context.Context, teamID uint) error {
 		return ctxerr.Wrap(ctx, err, "bulk set pending host profiles")
 	}
 
+	vc, ok := viewer.FromContext(ctx)
+	if !ok {
+		return fleet.ErrNoContext
+	}
+	filter := fleet.TeamFilter{User: vc.User, IncludeObserver: true}
+	hosts, err := svc.ds.ListHosts(ctx, filter, fleet.HostListOptions{TeamFilter: &teamID})
```

Comment:
Should the list of host ids be fetched before the team delete?

Reply:
awesome catch, thank you!